code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -54,7 +54,7 @@
 /obj/item/weapon/melee/energy/sword
 	name = "energy sword"
 	desc = "May the force be within you."
-	icon_state = "sword0"
+	icon_state = "sword0" //In the file it is called "sword", this is intentional so it does not show up in-hand when deactivated
```

Comment:
Dude, just null it

Reply:
Pretty sure it returns back to the icon_state in update_icon() because it checks for the "base state" and also whether it's active